ni_measurement_plugin_sdk_service/_internal/parameter/message_serializer.py

```diff
@@ -0,0 +1,238 @@
+# enums and default values
+from enum import Enum
+from typing import Any, Dict, Sequence
+from uuid import uuid4
+
+from google.protobuf import descriptor_pb2, descriptor_pool, message_factory, type_pb2
+from google.protobuf.descriptor_pb2 import FieldDescriptorProto
+
+# metadata
+from ni_measurement_plugin_sdk_service._internal.parameter.metadata import (
+    ParameterMetadata,
+)
+from tests.utilities.stubs.loopback.types_pb2 import ProtobufColor
+
+
+def serialize_parameters(
+    parameter_metadata_dict: Dict[int, ParameterMetadata],
+    parameter_values: Sequence[Any],
+) -> bytes:
+    """Serialize the parameter values in same order based on the metadata_dict.
+
+    Args:
+        parameter_metadata_dict (Dict[int, ParameterMetadata]): Parameter metadata by ID.
+
+        parameter_value (Sequence[Any]): Parameter values to serialize.
+
+    Returns:
+        bytes: Serialized byte string containing parameter values.
+    """
+    pool = descriptor_pool.Default()
+    file_descriptor_proto = descriptor_pb2.FileDescriptorProto()
+    original_guid = uuid4()
+    new_guid = "msg" + "".join(filter(str.isalnum, str(original_guid)))[:16]
```

Comment:
Add a str() around all of this so we don't have to do it multiple times below. I would rename the variable, e.g. `unique_descriptor_name`